clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1878,6 +2239,29 @@ private boolean updateFetchPositions(final long timeoutMs) {
         return true;
     }
 
+    /**
+     * Set the fetch position to the committed position (if there is one) 
+     * or reset it using the offset reset policy the user has configured
+     * within a given time limit.
+     * 
+     * @param start        the time at which the operation begins
+     * @param timeoutMs    the maximum duration of the method
+     * @throws org.apache.kafka.common.errors.AuthenticationException if authentication fails. See the exception for more details
+     * @throws NoOffsetForPartitionException If no offset is stored for a given partition and no offset reset policy is
+     *             defined
+     * @return true if all assigned positions have a position
+     */
+    private boolean updateFetchPositions(long start, long timeoutMs) {
```

Comment:
Can you explain why we cannot use the other `updateFetchPositions`?

Reply:
This was before John changed the old updateFetchPositions to take in a timeout argument. Before that, I had to improvise and I thought this was needed. However, this code is now irrevalant so I will now just switch over.